src/it_depends/resolution.py

```diff
@@ -304,17 +329,18 @@ def process_resolution(
                             elif isinstance(result, _DependencyResult):
                                 process_resolution(result.dep, result.packages, result.depth)
                             else:
-                                raise NotImplementedError(f"Unexpected future result: {result!r}")
+                                error_msg = f"Unexpected future result: {result!r}"
+                                raise NotImplementedError(error_msg)
 
     except KeyboardInterrupt:
         if sys.stderr.isatty() and sys.stdin.isatty():
             try:
                 while True:
                     sys.stderr.write("Would you like to output the partial results? [Yn] ")
                     choice = input().lower()
-                    if choice == "" or choice == "y":
+                    if choice in {"", "y"}:
```

Comment:
My intuition tells me this would be faster, but I may be wrong!
```suggestion
                    if choice in ("", "y"):
```

Reply:
You're right, I assumed that there was some time in instantiating the set (hashing the elements, etc.), particularly when the set is small (cardinality 2) and the strings are tiny 🤯 
```
>>> timeit.timeit(stmt="'n' in {'','y'}", number=100000);
0.0036437090020626783
>>> timeit.timeit(stmt="'n' in ('','y')", number=100000);
0.005011209053918719
```